### PR TITLE
Split EXPERIMENTAL_Table docs on multiple files

### PR DESCRIPTION
**Pull request by matheusps**
#### What is the purpose of this pull request?

Most of the work done in this PR was to move the docs on `README.md` to its own dedicated file. This was done to:

- ♻️Make the docs easier to maintain.
- ✨Make sections easier to reach.

👀Notable changes:
- `Columns` section  ➡️`Introduction.md` and was kept the same.
- `Features` section  ➡️`Components.md` and was kept the same.
- `State Handlers` and `Table Props` sections  ➡️ `Features.md` as got some new examples.
- `Migration Guide` section  ➡️`MigrationGuide.md` and was kept the same.
- `E2ETesting` section  ➡️`Testing.md` and was kept the same.

#### How should this be manually tested?

Clone the repo and `yarn && yarn start`

#### Screenshots or example usage

On the navigation menu: 
Screen Shot 2020-03-05 at 06 43 20

#### Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/EXPERIMENTAL_Table/docs/Features.md [hunk 1]
```diff
@@ -0,0 +1,561 @@
+### Measures
+
+The table measures are the current density, rowHeight, tableHeight and setCurrentDensity. They are calculated by the `useTableMeasures` hook.
```
Comment (matheusps) [inline react/components/EXPERIMENTAL_Table/docs/Features.md]: Added this description

File: react/components/EXPERIMENTAL_Table/docs/Features.md [hunk 2]
```diff
@@ -0,0 +1,561 @@
...
+function MeasuresExample() {
+  const measures = useTableMeasures({ size: items.length })
+  return <Table measures={measures} columns={columns} items={items} />
+}
+;<MeasuresExample />
+```
+
+### Proportional columns
+
+In some cases, the columns must be proportional within a ratio. This is achievable through the `useTableProportion` hook.
```
Comment (matheusps) [inline react/components/EXPERIMENTAL_Table/docs/Features.md]: Added this description

File: react/components/EXPERIMENTAL_Table/docs/Features.md [hunk 3]
```diff
@@ -0,0 +1,561 @@
...
+      <Table measures={measures} columns={sizedColumns} items={items} />
+    </>
+  )
+}
+;<ProportionExample />
+```
+
+### Toggle columns visibility
+
+The column's visibility can be toggled using `useTableVisibility`. This is useful to hide optional properties, providing a more distinct visualization.
```
Comment (matheusps) [inline react/components/EXPERIMENTAL_Table/docs/Features.md]: Added this description

File: react/components/EXPERIMENTAL_Table/docs/Features.md [hunk 4]
```diff
@@ -0,0 +1,561 @@
...
+      </Table>
+    </>
+  )
+}
+;<VisibilityExample />
+```
+
+### Sortable columns
+
+The `useTableSort` hook is designed to easily maintain columns sorting state. You may pass its return through the `sorting` prop.
```
Comment (matheusps) [inline react/components/EXPERIMENTAL_Table/docs/Features.md]: Added this description

File: react/components/EXPERIMENTAL_Table/docs/Features.md [hunk 5]
```diff
@@ -0,0 +1,561 @@
...
+      <Table.ActionBar>
+        <Toggle
+          label="Show custom loading"
+          checked={custom}
+          onChange={() => setCustom(custom => !custom)}
+        />
+      </Table.ActionBar>
+    </Table>
+  )
+}
```
Comment (matheusps) [inline react/components/EXPERIMENTAL_Table/docs/Features.md]: New example

File: react/components/EXPERIMENTAL_Table/docs/Features.md [hunk 6]
```diff
@@ -0,0 +1,561 @@
...
+          label="Show custom empty state"
+          checked={custom}
+          onChange={() => setCustom(custom => !custom)}
+        />
+      </Table.ActionBar>
+    </Table>
+  )
+}
+;<EmptyExample />
+```
```
Comment (matheusps) [inline react/components/EXPERIMENTAL_Table/docs/Features.md]: New example

File: react/components/EXPERIMENTAL_Table/docs/Features.md [hunk 7]
```diff
@@ -0,0 +1,561 @@
...
+        {
+          id: 'location',
+          title: 'Location',
+        },
+      ]}
+    />
+  )
+}
+;<ClickableExample />
+```
```
Comment (matheusps) [inline react/components/EXPERIMENTAL_Table/docs/Features.md]: New example

File: react/components/EXPERIMENTAL_Table/docs/Features.md [hunk 8]
```diff
@@ -0,0 +1,561 @@
...
```
Comment (matheusps) [inline react/components/EXPERIMENTAL_Table/docs/Features.md]: New example

File: styleguide.config.js [hunk 9]
```diff
@@ -164,7 +165,6 @@ module.exports = {
           components: [
             'react/components/EXPERIMENTAL_Conditions/index.js',
             'react/components/EXPERIMENTAL_Select/index.js',
-            'react/components/EXPERIMENTAL_Table/index.tsx',
```
Comment (matheusps) [inline styleguide.config.js]: Removed `Table` from here

File: styleguide.config.js [hunk 10]
```diff
@@ -179,6 +179,33 @@ module.exports = {
...
+                  content:
+                    'react/components/EXPERIMENTAL_Table/docs/Testing.md',
+                },
+                {
+                  name: '📚Migration Guide',
+                  content:
+                    'react/components/EXPERIMENTAL_Table/docs/MigrationGuide.md',
+                },
+              ],
+            },
```
Comment (matheusps) [inline styleguide.config.js]: To create this section

File: react/components/EXPERIMENTAL_Table/docs/Features.md [hunk 11]
```diff
@@ -0,0 +1,561 @@
...
+| Property | Type    | Required | Default         | Description               |
+| -------- | ------- | -------- | --------------- | ------------------------- |
+| size     | number  | ✅       | 0               | Length of the table items |
+| density  | Density | 🚫       | Density.Regular | Initial Density           |
+
+##### useTableMeasures Outputs
+
+| Property          | Type                       | Description               |
+| ----------------- | -------------------------- | ------------------------- |
+| currentDensity    | Density                    | The table current density |
```
Comment (rerissondaniel) [inline react/components/EXPERIMENTAL_Table/docs/Features.md]: Maybe this could be renamed to just `density` before the new major... But just a thought

Comment (matheusps) [reply react/components/EXPERIMENTAL_Table/docs/Features.md]: yes, it definitely could. would be cleaner

File: styleguide.config.js [hunk 12]
```diff
@@ -179,6 +179,33 @@ module.exports = {
...
+                  content:
+                    'react/components/EXPERIMENTAL_Table/docs/Features.md',
+                },
+                {
+                  name: 'Components',
+                  content:
+                    'react/components/EXPERIMENTAL_Table/docs/Components.md',
+                },
+                {
+                  name: '🧪E2E Testing',
```
Comment (kevinch) [inline styleguide.config.js]: I'd drop the emojies but that's really my opinion. Great job otherwise.

Comment (matheusps) [reply styleguide.config.js]: I'll do that. Thanks :)

**Pull request merged**